### PR TITLE
Subplugin register/unregister refactored. @open sesame 1/31 19:52

### PR DESCRIPTION
**Pull request by myungjoo**
A plugin, "filter" and "decoder" no more have their own
internal data structure to store subplugins.
    
Both filter and decoder subplugins now use APIs of
nnstreamer_subplugin via
nnstreamer_plugin_api.h (filters)
tensordec.h (decoder)
    
Decoders are to be refactored further to use
nnstreamer_plugin_api.h after moving decoder subplugins
to ext/
    
Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>

Wait for #1090 #1091

**Self evaluation:**
1. Build test: [x]Passed [ ]Failed [ ]Skipped
2. Run test: [x]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

File: gst/nnstreamer/tensor_filter/tensor_filter.c [hunk 1]
```diff
@@ -110,58 +110,15 @@
...
-    list = list->next;
-  } while (list != NULL);
-
-  GST_INFO ("A new sub-plugin, \"%s\" is registered for tensor_filter.\n",
-      tfsp->name);
-
-  /** @todo @bug unregister at exit */
-  hold_register_subplugin (NNS_SUBPLUGIN_FILTER, tfsp->name, tfsp);
-
+  register_subplugin (NNS_SUBPLUGIN_FILTER, tfsp->name, tfsp);
```
Comment (wooksong) [inline gst/nnstreamer/tensor_filter/tensor_filter.c]: tensor_filter_probe always returns TRUE whatever register_subplugin returns. How about returning the result of register_subplugin for the return value of the probe function.

File: gst/nnstreamer/tensor_filter/tensor_filter.c [hunk 2]
```diff
@@ -110,58 +110,15 @@
...
-};
-static TensorFilterSPList listhead = {.next = NULL,.body = &unknown };
-
 /**
  * @brief Filter subplugin should call this to register itself
  * @param[in] tfsp Tensor-Filter Sub-Plugin to be registered
  * @return TRUE if registered. FALSE is failed or duplicated.
  */
-gboolean
+int
```
Comment (wooksong) [inline gst/nnstreamer/tensor_filter/tensor_filter.c]: Q. Is this for cutting the dependency on glib? In the case of tensordec_probe, it still returns gboolean.

File: gst/nnstreamer/tensor_filter/tensor_filter.c (hunk 1 above)
Comment (myungjoo) [reply gst/nnstreamer/tensor_filter/tensor_filter.c]: Yup. it returns FALSE if duplicated. Will update.

File: gst/nnstreamer/tensor_filter/tensor_filter.c (hunk 2 above)
Comment (myungjoo) [reply gst/nnstreamer/tensor_filter/tensor_filter.c]: Yes. It's cutting dependency on glib for externally built subplugins (those who are in ext/ may choose not to use glib although our subplugins still use glib.)

The equivalent modifications for tensordec is in later PR (not the scope of this PR)

**Pull request merged**